sparse_strips/vello_hybrid/src/gpu.rs

```diff
@@ -0,0 +1,649 @@
+        );
+
+        let max_texture_dimension_2d = self.device.limits().max_texture_dimension_2d;
+        let config = Config {
+            width: self.surface_config.width,
+            height: self.surface_config.height,
+            strip_height: 4,
+            alpha_texture_width: max_texture_dimension_2d,
+        };
+
+        let alpha_len = render_data.alphas.len();
+        let alpha_texture_width = config.alpha_texture_width;
+        let alpha_texture_height =
+            (u32::try_from(alpha_len).unwrap()).div_ceil(alpha_texture_width);
+
+        let current_width = self.alphas_texture.width();
+        let current_height = self.alphas_texture.height();
+
+        if alpha_texture_width > current_width || alpha_texture_height > current_height {
+            // Ensure dimensions don't exceed WebGL2 limits
+            assert!(
+                alpha_texture_height <= max_texture_dimension_2d,
+                "Alpha texture height exceeds WebGL2 limit"
+            );
+
+            self.alphas_texture = self.device.create_texture(&wgpu::TextureDescriptor {
+                label: Some("Alpha Texture (Resized)"),
+                size: wgpu::Extent3d {
+                    width: alpha_texture_width,
+                    height: alpha_texture_height,
+                    depth_or_array_layers: 1,
+                },
+                mip_level_count: 1,
+                sample_count: 1,
+                dimension: wgpu::TextureDimension::D2,
+                format: wgpu::TextureFormat::R32Uint,
+                usage: wgpu::TextureUsages::TEXTURE_BINDING | wgpu::TextureUsages::COPY_DST,
+                view_formats: &[],
+            });
+            self.alphas_texture_view = self
+                .alphas_texture
+                .create_view(&wgpu::TextureViewDescriptor::default());
+            self.render_bind_group = self.device.create_bind_group(&wgpu::BindGroupDescriptor {
+                label: None,
+                layout: &self.render_bind_group_layout,
+                entries: &[
+                    wgpu::BindGroupEntry {
+                        binding: 0,
+                        resource: wgpu::BindingResource::TextureView(&self.alphas_texture_view),
+                    },
+                    wgpu::BindGroupEntry {
+                        binding: 1,
+                        resource: self.config_buf.as_entire_binding(),
+                    },
+                ],
+            });
+        }
+
+        // Prepare alpha data for the texture
+        let texture_width = self.alphas_texture.width();
+        let texture_height = self.alphas_texture.height();
+        let mut alpha_data = vec![0_u32; (texture_width * texture_height) as usize];
+
+        // Fill the buffer with alpha data
+        for (idx, alpha) in render_data.alphas.iter().enumerate() {
+            if idx < alpha_data.len() {
+                alpha_data[idx] = *alpha;
+            }
+        }
```

Comment:
Without digging into specifics, this looks an awful lot like [`slice::copy_from_slice`](https://doc.rust-lang.org/stable/std/primitive.slice.html#method.copy_from_slice) (indeed, I'm slightly surprised that clippy hasn't said anything)

Reply:
Oops, yep! `slice::copy_from_slice` can be used here, but also needed to align the source buffer with the texture size. Thank you!

Fixed in 2357a16